### PR TITLE
feat: Added IdentityService APIs

### PR DESCRIPTION
**Pull request by nicolatimeus**
> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

Adds a new set of APIs that allow to interact with Kura identities.

A first implementation will be added based on the current UserAdmin conventions.

The APIs also allows to define extensions that contribute additional configuration to Kura identities. The additional configuration is provided in the form of ComponentConfiguration. The extensions can then store/apply the additional configuration in an implementation defined manner (for example by storing additional  custom properties/credentials in the UserAdmin). Kura web UI and identity REST APIs will be modified to support managing the additional configuration.

### CODE REVIEW

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/identity/configuration/extension/package-info.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2020 Eurotech and/or its affiliates and others
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/identity/configuration/extension/package-info.java]: Should this be replaced? `2011, 2020` -> `2024`

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/identity/IdentityService.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+     *         part of the method call or {@code false} if the identity does not
+     *         exist.
+     * @throws KuraException if a failure occurs in deleting the identity.
+     */
+    public boolean deleteIdentity(final String identityName) throws KuraException;
+
+    /**
+     * Returns the configuration of all existing identities.
+     * 
+     * @param componentsToReturn the set of {@link IdentityConfigurationComponent}
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/identity/IdentityService.java:56]: Please better format here.

Missing `@return the` phrase.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/identity/IdentityService.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    public List<IdentityConfiguration> getIdentitiesConfiguration(
+            Set<Class<? extends IdentityConfigurationComponent>> componentsToReturn)
+            throws KuraException;
+
+    /**
+     * Returns the configuration of the identity with the given name.
+     * 
+     * @param identityName       the identity name.
+     * @param componentsToReturn the set of {@link IdentityConfigurationComponent}
+     *                           types to be
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/identity/IdentityService.java]: Please correct formatting

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/identity/IdentityService.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+     * method for an identity that has just been created with the
+     * {@link IdentityService#createIdentity(String)} method.
+     * 
+     * This
+     * method can be useful for example to allow a user interface to show the
+     * initial identity configuration to the user before creating it.
+     *
+     * @param identityName       the identity name.
+     * @param componentsToReturn the set of {@link IdentityConfigurationComponent}
+     *                           types to be
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/identity/IdentityService.java]: Please correct formatting

**Review by marcellorinaldo — APPROVED** [review]
Just minor style comments. LGTM

**Review by marcellorinaldo — APPROVED** [review]

**Comment by nicolatimeus** [comment]
Code coverage will be added in follow-up PRs, merging.

**Pull request merged**